returnsrcvr-checker/src/main/java/org/checkerframework/checker/framework/FrameworkSupport.java

```diff
@@ -0,0 +1,13 @@
+package org.checkerframework.checker.framework;
+
+
+import org.checkerframework.checker.returnsrcvr.ReturnsRcvrAnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+public interface FrameworkSupport {
+	/**
+	 * 
+	 * @return {@code true} if the method is generated by framework and returns {@code this} 
+	 */
+	boolean knownToReturnThis(AnnotatedTypeMirror.AnnotatedExecutableType t, ReturnsRcvrAnnotatedTypeFactory typeFactory);
```

Comment:
Let's not take the `ReturnsRcvrAnnotatedTypeFactory` as a parameter here.  Instead, either make the relevant methods in `ReturnsRcvrAnnotatedTypeFactory` into static methods, or move them into their own class for utility methods.

Reply:
I suggest moving those methods into a non-instantiable utility class, in the style of the `AnnotationUtils` class in the Checker Framework proper.

Reply:
It's actually caused by the following problem, I didn't find the way to make `hasAnnotation` static.

Reply:
@kelloggm I am not sure about what you suggest, like have two methods `knownToReturnThisInLombok` and `knownToReturnThisInAutoValue` in a `FrameworkSupportUtils` non-instantiable class?